packages/plugin-core/src/_extension.ts

```diff
@@ -98,6 +97,11 @@ import { WSUtils } from "./WSUtils";
 const MARKDOWN_WORD_PATTERN = new RegExp("([\\w\\.\\#]+)");
 // === Main
 
+/** Before sending saved telemetry events, wait this long (in ms) to make sure
+ * the workspace will likely remain open long enough for us to send everything.
+ */
+const DELAY_TO_SEND_SAVED_TELEMETRY = 15 * 1000;
```

Comment:
It could still be a problem if the user opens the workspace, we delete the file and send the telemetry, then the user closes the workspace before the telemetry is flushed.

If Segment/Amplitude can dedupe events, I could force a flush first then delete the files.

Reply:
The track() call returns a promise.  Can we do something like 

track(StoredTelemetryData).then(delete from file storage since we know it's uploaded). 

Also ok with current delay impl for now.